src/inputs/ObjectSearchCreate.tsx

```diff
@@ -69,21 +76,39 @@ class ObjectSearchCreate extends Component<IObjectSearchCreateProps> {
     return this.props.fieldConfig as IFieldConfigObjectSearchCreate;
   }
 
+  private get createFields(): IFieldConfig[] {
+    return getFieldSetFields(this.fieldConfig.createFields).map(createField => fillInFieldConfig(createField));
+  }
+
   private async onAddNew(search: string) {
     const { onAddNewToggle, formManager, fieldConfig } = this.injected;
     this.search = search;
 
-    formManager.form.setFieldsValue({ [fieldConfig.field]: {} }, () => {
-      this.isAddingNew.setTrue();
-      if (onAddNewToggle) {
-        onAddNewToggle(true);
-      }
-    });
+    const createFieldsToDefaults: FieldData[] = this.createFields.map(createField => ({
```

Comment:
so..`createFields` creates fields-ant fields with validation rules, label, editProps...etc., and `createFieldsToDefaults` overrides the `name` and assigns a default value to the field? Might be useful to have a short comment for each, since it's a bit hard to follow with both functions mapping over the `createFields`

Reply:
Hm, yeah, I could probably use some more help figuring out what to change, but is this better?

![image](https://user-images.githubusercontent.com/796717/141378166-4a5e134d-99d3-430c-ab7b-d881f657a7e1.png)